### PR TITLE
First branch

### PR DESCRIPTION
**Pull request by Marwen-Brini**
I created a bubble sort algorithm where each element is compared to the one next to it and they are swapped if the one on the left is larger than the one on the right. This continues until the array is eventually sorted.

### CODE REVIEW

File: index.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  array
+end
+def bubble_sort_by(arr)
+  (0..arr.length - 2).each do |i|
+    sort_it = yield(arr[i], arr[i + 1])
+    arr[i], arr[i + 1] = arr[i + 1], arr[i] if sort_it.positive?
+  end
+  arr
+end
+puts bubble_sort_by(%w[hi hello hey]) { |left, right| left.length - right.length }
```
Comment (pacyL2K19) [inline index.rb:22]: Please just return the array, no need to print them, this will be used only for debugging

**Review by pacyL2K19 — CHANGES_REQUESTED** [review]
# Status: Changes requiered :stop_button: 

Hello **Team** great job made so far, everything works and the code looks clean 
Just some changes are needed before getting approved for this project:
- [x] The readme file doesn't explain anything about the actual project, feel free to use this [Microverse Readme template](https://github.com/microverseinc/readme-template) to make it look professional
- [ ] Please make sure the description of the PR is more descriptive than what you provided to help the reviewer make tests for validation
- [ ] Please rename the entry point of the project to `bubble_sort.rb` instead of `index.rb` to help the next reviewer to execute automated tests of the project
- [ ] Kindly remove the debugging helpers from your project see detail bellow :arrow_down: 

Work on these small changes before you submit another code review request and feel free to contact on slack or [LinkedIn](https://www.linkedin.com/in/pacifique-linjanja/) for more details about the review 
Happy coding :smiley:

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
# Status: Request Changes image

I can see you worked hard to implement what the previous TSE requested 
but you still need to fix a few things and you are ready to take off image


- You still have rubocop errors try to fix that

- Write a good PR, a meaningful title, and description start with a description of the project and then features you worked on it looks more clear and professional

## Happy coding 

Regards,

Lydia Ingabire

**Review by pacyL2K19 — APPROVED** [review]
# Status: Approved :medal_sports: 

Hello **Team** Congratulations :tada: , your project looks cleaner now 
The project meets all the requirement and code linters pass 
 Feel free to merge this branch now and go-ahead to the next project 
In the next PR you will create, please be more descriptive in the message you put 
Happy coding for the next project

**Pull request merged**